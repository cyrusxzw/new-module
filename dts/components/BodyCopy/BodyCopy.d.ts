import React from 'react';
import { BodyCopyProps } from './BodyCopy.types';
declare const BodyCopy: React.ForwardRefExoticComponent<
  BodyCopyProps & React.RefAttributes<HTMLElement>
>;
export { BodyCopy };
