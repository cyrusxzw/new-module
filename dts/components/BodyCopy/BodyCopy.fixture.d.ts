/// <reference types="react" />
declare const BodyCopyFixture: {
  eyebrow: string;
  heading: string;
  subHeading: string;
  theme: string;
  linkButtonGroup: {
    textAlign: string;
    children: JSX.Element;
  };
  links: {
    links: {
      id: string;
      style: string;
      children: string;
      url: string;
    }[];
  };
  copy: string;
};
export { BodyCopyFixture };
