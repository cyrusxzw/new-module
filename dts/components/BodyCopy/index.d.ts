import { BodyCopy } from './BodyCopy';
export { BodyCopy };
