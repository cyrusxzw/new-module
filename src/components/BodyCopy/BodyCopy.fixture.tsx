import React from 'react';
import { HYPERLINK_STYLE_TYPES } from '~/constants';
import { Hyperlink } from '~/components/Hyperlink';

const {
  EXTERNAL_NO_ICON_TEXT_LINK,
  EXTERNAL_NO_ICON_BUTTON_LINK,
} = HYPERLINK_STYLE_TYPES;

const BodyCopyFixture = {
  eyebrow: 'Nullam id erat est',
  heading: 'Lorem ipsum dolor sit amet',
  subHeading: 'Nunc dignissim quis arcu a',
  theme: 'dark',
  linkButtonGroup: {
    textAlign: 'left',
    children: (
      <>
        <Hyperlink style={EXTERNAL_NO_ICON_BUTTON_LINK} url="/">
          Proin vulputate dignissim
        </Hyperlink>
        <Hyperlink style={EXTERNAL_NO_ICON_BUTTON_LINK} url="Quisque lacus">
          Quisque lacus dolor
        </Hyperlink>
      </>
    ),
  },
  links: {
    links: [
      {
        id: 'Morbi-id-nulla',
        style: EXTERNAL_NO_ICON_TEXT_LINK,
        children: 'Morbi id nulla',
        url: 'https://aesop.com/',
      },
      {
        id: 'Nullam-ultrices',
        style: EXTERNAL_NO_ICON_TEXT_LINK,
        children: 'Nullam ultrices',
        url: 'https://aesop.com/',
      },
    ],
  },

  copy:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam id erat est. Ut convallis, neque id semper fermentum, urna risus gravida nibh, vel molestie sapien leo ac tortor. Nunc dignissim quis arcu a lobortis. Phasellus tempus consequat erat, at egestas elit pulvinar lacinia. Nullam pretium, risus a dictum rhoncus, ex libero cursus erat, dignissim malesuada tortor justo quis enim.',
};

export { BodyCopyFixture };
