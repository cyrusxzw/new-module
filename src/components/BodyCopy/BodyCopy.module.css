@import 'styles/index';

.base {
  font-family: var(--font-family-sans-serif-regular);

  &:not(:last-child) {
    margin-bottom: var(--layout-sm-spacing);
  }

  @media (--viewport-md) {
    &:not(:last-child) {
      margin-bottom: var(--layout-md-spacing);
    }
  }

  @media (--viewport-lg) {
    &:not(:last-child) {
      margin-bottom: var(--layout-lg-spacing);
    }
  }
}

.contentWrapper {
  margin-top: 30px;
  font-size: rem(16px);

  a {
    font-family: var(--font-family-sans-serif-medium);
    font-size: rem(14px);
  }
}

.copy {
  &,
  p {
    font-size: rem(16px);
    line-height: 1.7;

    &:first-of-type {
      margin-top: 0;
    }

    &:last-of-type {
      margin-bottom: 0;
    }

    a {
      display: inline;
      font-weight: var(--font-weight-normal);
    }
  }
}

.dark {
  .copy {
    &,
    p {
      @extend %font-theme-dark-copy;
    }

    a {
      color: var(--color-dark-link);

      &:hover,
      &:focus {
        color: var(--color-dark-link-hover);
      }
    }
  }
}

.light.light {
  .copy {
    &,
    p {
      @extend %font-theme-light-copy;
    }

    a {
      color: var(--color-light-link);

      &:hover,
      &:focus {
        color: var(--color-light-link-hover);
      }
    }
  }
}
