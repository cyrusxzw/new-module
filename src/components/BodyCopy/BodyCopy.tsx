import React, { forwardRef } from 'react';
import cx from 'classnames';
import { useThemeContext } from '~/contexts';
import { SectionHeading } from '~/components/SectionHeading';
import { BodyCopyProps } from './BodyCopy.types';
import styles from './BodyCopy.module.css';

const BodyCopy = forwardRef<HTMLElement, BodyCopyProps>(
  (
    {
      childrenClassNames,
      className,
      content,
      copy,
      eyebrow,
      hasFlushHeading = false,
      hasSerifFontHeading = false,
      heading,
      id,
      isHeroArticle,
      subHeading,
      theme,
    },
    ref,
  ) => {
    const currentTheme = useThemeContext(theme, 'dark');
    const classSet = cx(styles.base, styles[currentTheme], className);

    return (
      <article className={classSet} id={id} ref={ref}>
        <SectionHeading
          childrenClassNames={{
            eyebrow: cx(styles.eyebrow, childrenClassNames?.eyebrow),
            heading: cx(styles.heading, childrenClassNames?.heading),
            subHeading: cx(styles.subHeading, childrenClassNames?.subHeading),
          }}
          eyebrow={eyebrow}
          hasSerifFontHeading={hasSerifFontHeading}
          heading={heading}
          id={id}
          isFlush={hasFlushHeading}
          isPageHeading={isHeroArticle}
          subHeading={subHeading}
          theme={currentTheme}
        />

        {copy && <div className={styles.copy}>{copy}</div>}

        {content && <div className={styles.contentWrapper}>{content}</div>}
      </article>
    );
  },
);

export { BodyCopy };
