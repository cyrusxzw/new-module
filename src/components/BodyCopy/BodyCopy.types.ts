import { ReactNode } from 'react';
import type { Themes } from '~/types';

type BodyCopyProps = {
  childrenClassNames?: {
    eyebrow?: string;
    heading?: string;
    subHeading?: string;
  };
  className?: string;
  content?: ReactNode;
  copy?: ReactNode;
  eyebrow?: string;
  hasFlushHeading?: boolean;
  hasSerifFontHeading?: boolean;
  heading?: string;
  id?: string;
  isHeroArticle?: boolean;
  subHeading?: string;
  theme?: Themes;
};

export type { BodyCopyProps };
