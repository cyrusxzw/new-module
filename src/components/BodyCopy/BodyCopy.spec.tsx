import React from 'react';
import { render } from '@testing-library/react';
import { BodyCopy } from './BodyCopy';
import { BodyCopyFixture } from './BodyCopy.fixture';

describe('<BodyCopy />', () => {
  it('should be defined', () => {
    expect(BodyCopy).toBeDefined();
  });

  it('should render base component correctly', () => {
    const { container } = render(
      <BodyCopy
        content={<div>{BodyCopyFixture.linkButtonGroup.children}</div>}
        copy="Parsley Seed Cream. [The Paris Review](http://theparisreview.org) Intensely soothing, nourishing and hydrating, this elegant formulation contains a potent blend of botanicals that offer fortification of the highest order against free radicals."
        eyebrow="Recommended Nearby"
        heading="Aesop & The Paris Review: A Partnership Extended"
        id="Anti-Oxidant"
        subHeading="Active Nutrients"
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('does not render content when no `content` prop supplied', () => {
    const { container } = render(
      <BodyCopy
        copy="Parsley Seed Cream. [The Paris Review](http://theparisreview.org) Intensely soothing, nourishing and hydrating, this elegant formulation contains a potent blend of botanicals that offer fortification of the highest order against free radicals."
        eyebrow="Recommended Nearby"
        heading="Aesop & The Paris Review: A Partnership Extended"
        id="Anti-Oxidant"
        subHeading="Active Nutrients"
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders promary title as an `h2` if no `eyebrow` prop is passed', () => {
    const { container } = render(
      <BodyCopy heading="Aesop & The Paris Review: A Partnership Extended" />,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders promary title as an `h2` if no `eyebrow` or `primaryTitle` prop are passed', () => {
    const { container } = render(
      <BodyCopy subHeading="Aesop & The Paris Review: A Partnership Extended" />,
    );

    expect(container).toMatchSnapshot();
  });
});
