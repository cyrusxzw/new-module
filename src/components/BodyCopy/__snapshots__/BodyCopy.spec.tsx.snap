// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BodyCopy /> does not render content when no \`content\` prop supplied 1`] = `
<div>
  <article
    class="base dark"
    id="Anti-Oxidant"
  >
    <header
      class="base"
      id="Anti-Oxidant"
    >
      <h2
        class="base xLarge dark heading heading"
      >
        Aesop & The Paris Review: A Partnership Extended
      </h2>
      <h3
        class="base mediumWeightFont xXSmall dark eyebrow eyebrow"
      >
        Recommended Nearby
      </h3>
      <h4
        class="base mediumWeightFont xSmall dark subHeading subHeading"
      >
        Active Nutrients
      </h4>
    </header>
    <div
      class="copy"
    >
      Parsley Seed Cream. [The Paris Review](http://theparisreview.org) Intensely soothing, nourishing and hydrating, this elegant formulation contains a potent blend of botanicals that offer fortification of the highest order against free radicals.
    </div>
  </article>
</div>
`;

exports[`<BodyCopy /> should render base component correctly 1`] = `
<div>
  <article
    class="base dark"
    id="Anti-Oxidant"
  >
    <header
      class="base"
      id="Anti-Oxidant"
    >
      <h2
        class="base xLarge dark heading heading"
      >
        Aesop & The Paris Review: A Partnership Extended
      </h2>
      <h3
        class="base mediumWeightFont xXSmall dark eyebrow eyebrow"
      >
        Recommended Nearby
      </h3>
      <h4
        class="base mediumWeightFont xSmall dark subHeading subHeading"
      >
        Active Nutrients
      </h4>
    </header>
    <div
      class="copy"
    >
      Parsley Seed Cream. [The Paris Review](http://theparisreview.org) Intensely soothing, nourishing and hydrating, this elegant formulation contains a potent blend of botanicals that offer fortification of the highest order against free radicals.
    </div>
    <div
      class="contentWrapper"
    >
      <div>
        <a
          class="base blockStyle left dark"
          href="/"
          target="_self"
        >
          Proin vulputate dignissim
        </a>
        <a
          class="base blockStyle left dark"
          href="Quisque lacus"
          target="_self"
        >
          Quisque lacus dolor
        </a>
      </div>
    </div>
  </article>
</div>
`;

exports[`<BodyCopy /> renders promary title as an \`h2\` if no \`eyebrow\` or \`primaryTitle\` prop are passed 1`] = `
<div>
  <article
    class="base dark"
  >
    <header
      class="base"
    >
      <h2
        class="base mediumWeightFont xSmall dark subHeading subHeading"
      >
        Aesop & The Paris Review: A Partnership Extended
      </h2>
    </header>
  </article>
</div>
`;

exports[`<BodyCopy /> renders promary title as an \`h2\` if no \`eyebrow\` prop is passed 1`] = `
<div>
  <article
    class="base dark"
  >
    <header
      class="base"
    >
      <h2
        class="base xLarge dark heading heading"
      >
        Aesop & The Paris Review: A Partnership Extended
      </h2>
    </header>
  </article>
</div>
`;
